pandas/core/generic.py

```diff
@@ -1940,6 +1940,84 @@ def tail(self, n=5):
+               Default = 5 if frac = None. 
+            frac: share of rows to return. Cannot be used with n. 
+            replace {True, False}: Sample with or without replacement.
+            weights: Accepts a Series, list, or ndarray of weights. If called on a 
+                    DataFrame, will also accept the name of a column as a string. 
+                    Must be same length as index.  
+                    Default 'None' results in equal probability weighting.
+            seed: seed to be fed to numpy random.RandomState() Function. Default None. 
+        -----------
+        Note: sample sizes are passed to numpy.random.choice(), which accepts negative values. 
+              
+              If a negative value is passed, this will be interpreted with respect to 
+              number of rows. So n = -1 will result in a sample with 
+              [original number of rows - 1] rows.
+              
+              A similar behavior applies to values passed to frac. A value of frac = -0.75 
+              will return a sample with a size of [0.25 * original number of rows] rows, 
+              give or take rounding. 
+              
+              Finally, note that non-integer arguments passed to n will be rounded off by numpy. 
+              So n = 3.1 will return 3 rows, and n = 3.9 rows will return 4 rows. 
+        """
+    
+        # Setup vars
+        rs = np.random.RandomState(seed)
+        length = len(self)
+    
+        # Check weights for compliance
+        if weights != None:
+                                      
+            # If series, return ndarray
+            if type(weights) == pd.core.series.Series:
+                weights = weights.values
+                
+            elif type(weights) == list:
+                pass
+    
+            # If this is a dataframe and we get a string, check for column. 
+            elif type(self) == pd.core.frame.DataFrame and (type(weights) == str or type(weights) == unicode):
+                try: 
+                    weights = self[weights].values
+                except: 
+                    raise ValueError("Weights string not valid column name")
+                
+            # If any other type cases arise, throw error. 
+            elif type(weights) != np.ndarray:
+                raise ValueError("Weights must be a Series, list, numpy array, or -- if dataframe -- string name of column")
+
+
+            # Check length (numpy does this, but has confusing errors with different argument labels.)
+            if len(weights) != length:
+                raise ValueError("Weights and data to be sampled must be of same length")
+
+    
+    
+        # Check whether frac or N
+        if n == None and frac == None:
+            n = 5
+        elif n == None and frac != None:
+            n = int(round(frac * length)) 
+        elif n != None and frac != None:
+            raise NotImplementedError('Please enter a value for "frac" OR "n", not both')
```

Comment:
This should be `ValueError`. `NotImplementedError` should be reserved for code that is actually incomplete.